packages/wrangler/src/ai/createApp.tsx

```diff
@@ -0,0 +1,69 @@
+import { Text, Box } from "ink";
+import React from "react";
+import { fetchResult } from "../cfetch";
+import { withConfig } from "../config";
+import { logger } from "../logger";
+import { requireAuth } from "../user";
+import { renderToString } from "../utils/render";
+import { constellationBetaWarning } from "./utils";
+import type {
+	CommonYargsArgv,
+	StrictYargsOptionsToInterface,
+} from "../yargs-types";
+import type { Application } from "./types";
+
+export function Options(yargs: CommonYargsArgv) {
+	return yargs
+		.positional("name", {
+			describe: "The name of the new Application",
+			type: "string",
+			demandOption: true,
+		})
+		.positional("runtime", {
+			describe: "The name of the runtime to use",
+			type: "string",
+			demandOption: true,
+		})
+		.epilogue(constellationBetaWarning);
+}
+
+type HandlerOptions = StrictYargsOptionsToInterface<typeof Options>;
+export const Handler = withConfig<HandlerOptions>(
+	async ({ name, runtime, config }): Promise<void> => {
+		const accountId = await requireAuth(config);
+
+		logger.log(constellationBetaWarning);
+
+		let app: Application;
+		try {
+			app = await fetchResult(
+				`/accounts/${accountId}/constellation/application`,
+				{
+					method: "POST",
+					headers: {
+						"Content-Type": "application/json",
+					},
+					body: JSON.stringify({
+						name,
+						runtime,
+					}),
+				}
+			);
+		} catch (e) {
+			if ((e as { code: number }).code === 7502) {
```

Comment:
7502 is a generic database error. I'll introduce a new error code for name collision.